DDP-5758 - removed unused code (replaced with new AI builder)

evgeniipr opened the pull request:
## Context

Refactored `ActivityInstanceFromDefinitionBuilder` module: added possibility to re-define any part of code via factories. 
It is created `AIBuilderFactory` which provides creation of AI builder and aggregates other kind of factories:
- `AICreatorsFactory`;
- `AIContentRendererFactory`;
- `TemplateRenderFactory`.
A default implementation of `AIBuilderFactory` passed to `ActivityInstanceFromDefinitionBuilder` during it's creation. And it's possible to implement a custom version of `AIBuilderFactory` and pass it as a parameter during creation of `ActivityInstanceFromDefinitionBuilder'. So, any part of code if AI builder could be customized.

Implemented unit tests instead of deleted one (testing deleted classes). 

Deleted classes which not needed and not used anymore (replaced with usage of `ActivityInstanceFromDefinitionBuilder`).
The following classes and files are deleted:
- `FormInstanceDao.java`,
- `SectionBlockDao.java`,
- `JdbiFormActivityFormSection.sql.stg`;
- `SectionBlockDaoTest.java`.

Implemented unit test instead of the deleted one: `AIBuilderTest.java`.


## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

Commit:
DDP-5758 - refactored ActivityInstanceFromDefinitionBuilder module: added possibility to re-define any part of code via factories; implemented unit tests instead of deleted one (testing deleted classes); deleted classes which not needed and not used anymore (replaced with usage of ActivityInstanceFromDefinitionBuilder)

Commit:
DDP-5758 - refactor utility classes: convert it to factories, so it is possible to override it

Commit:
DDP-5758 - added comments to new factory classes

Commit:
DDP-5758 - rename method to more good-looking name

yufengwng commented on pepper-apis/src/test/java/org/broadinstitute/ddp/db/FormInstanceDaoTest.java:
```diff
@@ -41,13 +41,10 @@
-        SectionBlockDao sectionBlockDao = new SectionBlockDao();
-        dao = FormInstanceDao.fromDaoAndConfig(sectionBlockDao, sqlConfig);
-        org.broadinstitute.ddp.db.ActivityInstanceDao activityInstanceDao = new org.broadinstitute.ddp.db.ActivityInstanceDao(dao);
+        org.broadinstitute.ddp.db.ActivityInstanceDao activityInstanceDao = new org.broadinstitute.ddp.db.ActivityInstanceDao();
```
Should we move these tests into the test file for `ActivityInstanceService` and then get rid of this `FormInstanceDaoTest` file?

evgeniipr replied:
Done

yufengwng reviewed (dismissed):
🎉   Woohoo! Awesome! We're finally removing `FormInstanceDao` and `SectionBlockDao`!

Commit:
DDP-5758 - remove FormInstanceDaoTest (move all tests to ActivityInstanceServiceTest) as class FormInstanceDao was removed too

Commit:
DDP-5758 - refactor FormInstanceTest; remove unised code from FormInstance; reorganise AI Builder module structure; remove validation rule message detection code from ActivityDefStore

yufengwng reviewed (dismissed):
I didn't think we need so many "services" files... but I guess it's fine for now. Next time, let's try to keep those factories/creators/helpers/services to a minimum. There's way too many things and layers of indirection to keep track of.

Commit:
DDP-5758 - remove services and other small classes (join into bigger ones) - try to make structure of AI builder module more clear

evgeniipr commented:
> I didn't think we need so many "services" files... but I guess it's fine for now. Next time, let's try to keep those factories/creators/helpers/services to a minimum. There's way too many things and layers of indirection to keep track of.

@yufengwng I removed services, tried to make code more clear.. The Basic idea is: all classes building the ActivityInstance could be possible to customize (via AIBuilderFactory) - so provide code flexibility and loose coupling. Now I removed all other services, factories, leaving only AIBuilderFactory. And the methods from services moved to corresponding Creator-classes. I agree that 'Service' - was not a good name for such kind of classes. 
Please re-approve (this refactoring is quite safe: just moving methods from one classes to other).

Commit:
DDP-5758 - rename class, collect all template-rendering related methods in one class

evgeniipr committed:
Merge branch 'develop' into DDP-5758-remove-unused-code

yufengwng reviewed (approved):
Looks good-to-go.
